docs/source/API/core/parallel-dispatch/UniqueToken.rst

```diff
@@ -0,0 +1,71 @@
+``UniqueToken``
+===============
+
+.. role:: cppkokkos(code)
+    :language: cppkokkos
+
+Header File: ``Kokkos_Core.hpp``
+
+
+Description
+------------
+
+``UniqueToken`` is a portable way to acquire a unique ID for calling a thread (``thread-id`` is not portable execution environments).  ``UniqueToken`` is thus analogous to ``thread-id``, and has a ``UniqueTokenScope`` template parameter (default: ``Instance``, but can be ``Global``).    
+
+Interface
+---------
+
+.. cppkokkos:class:: template <class ExecutionSpace, UniqueTokenScope :: Global> UniqueToken
+
+
+
+Parameters
+-----------
+
+*  ``ExecutionSpace``:  See `Execution Spaces <../execution_spaces.html>`_
+
+.. note::
+   In a parallel region, before the main computation, a pool of ``UniqueToken`` (integer) Id is generated, and each Id is released following iteration.
+
+.. warning::
+   ``UniqueToken <ExecutionSpace> token`` *can* be called inside a parallel region, *but* must be released at the end of *each* iteration.
+
+
+*  ``UniqueTokenScope``:  defaults to ``Instance``, but ``Global`` can be employed when thread awareness is needed for more than one ``ExecutionSpace`` instance, as in the case of submitting concurrent kernels to CUDA streams.
+
+
+Constructors
+-------------
+  .. cppkokkos:function:: UniqueToken(size_t max_size, ExecutionSpace execution_space, UniqueTokenScope :: Global)
+
+
+
+
```

Comment:
```suggestion
Constructors
-------------
  .. cppkokkos:function:: UniqueToken(ExecutionSpace execution_space = ExecutionSpace())

Additionally for ``UniqueTokenScope==Instance``
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  .. cppkokkos:function:: UniqueToken(size_t max_size, ExecutionSpace execution_space=ExecutionSpace())
     Sets an upper bound to the number of tokens. Requires ``max_size <= execution_space.concurrency()``
     
Token Interface
---------------
 .. cppkokkos:function:: size_type size()
     Returns the size of the token pool
     
 .. cppkokkos:function:: size_type acquire()
     Returns the token for the executing tread
     
 .. warning::
     Acquired tokens *must* be released at the end of the parallel region in which they were acquired.

 .. cppkokkos:function:: void release(size_type idx)
     Releases the passed token
```

@crtrott can you verify that it is ``max_size <= execution_space.concurrency()`` ... the comment in https://github.com/kokkos/kokkos/blob/d4af7c8302ecb4280c6215497bb2e39f9ed83c3f/core/src/Kokkos_UniqueToken.hpp#L78 would suggest otherwise, but the following lines suggest the opposite. But I might be wrong here :-D